components/profile_popover/profile_popover.jsx

```diff
@@ -257,17 +260,18 @@ class ProfilePopover extends React.PureComponent {
     };
 
     renderCustomStatus() {
-        const {customStatus, isCustomStatusEnabled, user, currentUserId, hideStatus} = this.props;
+        const {customStatus, isCustomStatusEnabled, user, currentUserId, hideStatus, isCustomStatusExpired} = this.props;
 
-        const customStatusSet = (customStatus.text || customStatus.emoji);
+        const customStatusSet = (customStatus?.text || customStatus?.emoji) && !isCustomStatusExpired;
         const canSetCustomStatus = (user.id === currentUserId);
-        const shouldShowCustomStatus = isCustomStatusEnabled && !hideStatus && customStatus && (customStatusSet || canSetCustomStatus);
+        const shouldShowCustomStatus = isCustomStatusEnabled && !hideStatus && (customStatusSet || canSetCustomStatus);
 
         if (!shouldShowCustomStatus) {
-            return null;
+            return {};
```

Comment:
why did you change from null to an empty object?

Reply:
So that I could use destructuring like this... 
`const {customStatusContent, expiryContent} = this.renderCustomStatus();`
without checking for null.